### PR TITLE
add patch and integration CI dimension for sslproxy

### PR DESCRIPTION
**Pull request by samuel40791765**
### Issues:
Resolves `CryptoAlg-1962`

### Description of changes: 
We've recently ironed out some of the gaps for SSLProxy support. Most of the gaps were related to testing configurations.
The gaps were:
* The `prime192v1` being used within tests (not supported in AWS-LC).
* The `TMP_SESS_FILE` used for `SSL_Session` caching in files was only relevant to OpenSSL and LibreSSL. The patch I've included a new pem file specific to AWS-LC.

This adds a CI dimension to verify our newfound support.

### Call-outs:
We'll be looking to contribute this patch to SSLproxy some time soon.

### Testing:
New CI dimension

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: tests/ci/integration/run_sslproxy_integration.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+#    - SCRATCH_FOLDER
+#      - sslproxy
+#      - AWS_LC_BUILD_FOLDER
+#      - AWS_LC_INSTALL_FOLDER
+#      - SSLPROXY_BUILD_FOLDER
+#      - libevent
+#      - LIBEVENT_INSTALL_FOLDER
+
+# Assumes script is executed from the root of aws-lc directory
+SCRATCH_FOLDER=${SRC_ROOT}/"SSLPROXY_BUILD_ROOT"
```
Comment (skmcgrail) [inline tests/ci/integration/run_sslproxy_integration.sh:20]: I believe the best practice is to have this be `"${SRC_ROOT}/SSLPROXY_BUILD_ROOT"`. (variable should be inside the quotes). There are a few other lines where the two styles are mixed as well.

Comment (samuel40791765) [reply tests/ci/integration/run_sslproxy_integration.sh:20]: Will align and check the other NGINX PR while I'm at it as well.

File: tests/ci/integration/run_sslproxy_integration.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+AWS_LC_BUILD_FOLDER="${SCRATCH_FOLDER}/aws-lc-build"
+AWS_LC_INSTALL_FOLDER="${SCRATCH_FOLDER}/aws-lc-install"
+LIBEVENT_SRC_FOLDER="${SCRATCH_FOLDER}/libevent"
+LIBEVENT_INSTALL_FOLDER="${SCRATCH_FOLDER}/libevent-install"
+
+mkdir -p ${SCRATCH_FOLDER}
+rm -rf ${SCRATCH_FOLDER}/*
+cd ${SCRATCH_FOLDER}
+
+function libevent_install() {
```
Comment (skmcgrail) [inline tests/ci/integration/run_sslproxy_integration.sh:33]: I never knew libevent dependened on / supported OpenSSL specific features.

**Review by skmcgrail — DISMISSED** [review]

File: tests/ci/integration/sslproxy_patch/aws-lc-sslproxy.patch [hunk 3]
```diff
@@ -0,0 +1,115 @@
+diff --git a/tests/check/cachedsess.t.c b/tests/check/cachedsess.t.c
+index 7dbac6e..58a47d5 100644
+--- a/tests/check/cachedsess.t.c
++++ b/tests/check/cachedsess.t.c
+@@ -40,6 +40,8 @@
+ 
+ #if defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x20501000L
+ #define TMP_SESS_FILE "pki/session-libressl-2.5.0.pem"
++#elif defined(AWSLC_API_VERSION) && AWSLC_API_VERSION == 22
```
Comment (andrewhop) [inline tests/ci/integration/sslproxy_patch/aws-lc-sslproxy.patch:9]: Why the specific check for API version 22? Did we add support in version 23? If so should this be <= 22?

Comment (samuel40791765) [reply tests/ci/integration/sslproxy_patch/aws-lc-sslproxy.patch:9]: `22` is the current version we're on. This test tests against the dumped out contents of `SSL_SESSION` and checks if it's reparsable. This isn't standardized behavior however, and we don't make any promises that contents `SSL_SESSION` won't change. The test understands this and only compares against specific versions of libcrypto.
* https://github.com/sonertari/SSLproxy/blob/30ed5b420002c8f1ac6f5c865373baa1e46bc4b8/tests/check/cachedsess.t.c#L41

I'm setting the specific `AWSLC_API_VERSION` here to help the test detect incompatibility, whenever we update the contents of our `SSL_SESSION`. There isn't a guarantee that older versions of AWS-LC are compatible with the current `session-aws-lc-1.15.0.pem` file, which is why I used `== 22`.

File: tests/ci/integration/run_sslproxy_integration.sh (hunk 2 above)
Comment (samuel40791765) [reply tests/ci/integration/run_sslproxy_integration.sh:33]: Yeah me neither. Apparently it's configurable, and they support LibreSSL as well.

**Review by skmcgrail — APPROVED** [review]

File: tests/ci/integration/sslproxy_patch/aws-lc-sslproxy.patch (hunk 3 above)
Comment (andrewhop) [reply tests/ci/integration/sslproxy_patch/aws-lc-sslproxy.patch:9]: Do you have a mechanism to remember this? I'm concerned we'll bump our API version for some other reason which will turn off this test. Then in the future we could forget to update this and/or the test.

Also if we do remember to update this test is it documented how to generate this test file?

Comment (samuel40791765) [reply tests/ci/integration/sslproxy_patch/aws-lc-sslproxy.patch:9]: Added `sslproxy_patch_reminder` in the bash script to remind us, along with steps to regenerate the test file.

**Review by andrewhop — APPROVED** [review]

**Review by skmcgrail — APPROVED** [review]

**Pull request merged**